src/enveloping/Commitment.ts

```diff
@@ -2,56 +2,63 @@ import { Address } from '../relayclient/types/Aliases'
 import { PrefixedHexString } from 'ethereumjs-tx'
 import { ethers } from 'ethers'
 
-export interface CommitmentReceipt {
-  commitment: Commitment
-  workerSignature: PrefixedHexString
-  workerAddress: Address
-}
-
 export interface CommitmentResponse {
   signedTx: PrefixedHexString
   signedReceipt?: CommitmentReceipt
   transactionHash: PrefixedHexString
 }
 
+export interface CommitmentReceipt {
+  commitment: Commitment
+  workerSignature: PrefixedHexString
+  workerAddress: Address
+}
+
 export class Commitment {
   time: Number
   from: Address
   to: Address
   data: PrefixedHexString
   relayHubAddress: Address
   relayWorker: Address
+  enableQos: boolean
+  signature: PrefixedHexString
 
   constructor (
     time: Number,
     from: Address,
     to: Address,
     data: PrefixedHexString,
     relayHubAddress: Address,
-    relayWorker: Address
+    relayWorker: Address,
+    enableQos: boolean,
+    signature: PrefixedHexString
   ) {
     this.time = time
     this.from = from
     this.to = to
     this.data = data
     this.relayHubAddress = relayHubAddress
     this.relayWorker = relayWorker
+    this.enableQos = enableQos
+    this.signature = signature
   }
 
-  public static TypeEncoding = 'commit(uint,address,address,bytes,address,address)'
+  public static encodedFieldTypes = ['uint256', 'address', 'address', 'bytes', 'address', 'address', 'bool']
 
-  orderForEnc (): any[] {
+  encodedFields (): any[] {
     return [
       this.time,
       this.from,
       this.to,
       this.data,
       this.relayHubAddress,
-      this.relayWorker
+      this.relayWorker,
+      this.enableQos
     ]
   }
 
-  encodeForSign (relayHubAddress: Address): PrefixedHexString {
-    return ethers.utils.defaultAbiCoder.encode([Commitment.TypeEncoding, 'address'], [this.orderForEnc(), relayHubAddress])
+  encodeForSign (): PrefixedHexString {
```

Comment:
the previous encoding was incorrect as it received the hub address as parameter and as a result encoded this field twice

Reply:
Good catch here